metrics/metrics.go

```diff
@@ -72,33 +73,33 @@ func ensureNotFrozen() {
 	}
 }
 
-// RegisterRootScope initializes the root scope for all the service metrics
-func RegisterRootScope(rep ScopeFunc) {
+// RegisterStatsReporter initializes the stats reporter for all the service metrics
+func RegisterStatsReporter(repFunc StatsReporterFunc) {
 	ensureNotFrozen()
 
 	_mu.Lock()
 	defer _mu.Unlock()
 
-	if _scopeFunc != nil {
+	if _repFunc != nil {
 		// TODO(glib): consider a "force" flag, or some way to clear out and replace the reporter
 		panic("There can be only one metrics reporter")
 	}
 
-	_scopeFunc = rep
+	_repFunc = repFunc
 }
 
 // RootScope returns the provided stats reporter, or nil
-func RootScope(i ScopeInit) (tally.Scope, io.Closer) {
+func RootScope(i ScopeInit) (tally.Scope, tally.StatsReporter, io.Closer) {
 	_mu.Lock()
 	defer _mu.Unlock()
 
-	if _scopeFunc != nil {
-		rep, closer, err := _scopeFunc(i)
+	if _repFunc != nil {
+		rep, err := _repFunc(i)
 		if err != nil {
 			panic(fmt.Sprintf("Failed to initialize metrics reporter %v", err))
 		}
-		return rep, closer
+		scope, closer := tally.NewRootScope("", nil, rep, time.Second)
```

Comment:
Service owners can still configure using WithMetrics options. From further conversation with Glib, he suggested we might want to add special configuration for uber internal services. For this reason, changed to allowing scope and reporter to be passed in and use the pre-configured metrics scope.